src/prob/pf_bf.jl

```diff
@@ -13,23 +13,38 @@ end
 ""
 function build_mc_pf_bf(pm::_PMs.AbstractPowerModel)
     # Variables
-    variable_mc_voltage(pm; bounded=false)
+    variable_mc_voltage(pm; bounded=true) #TODO should be false
     variable_mc_branch_current(pm)
     variable_mc_branch_flow(pm)
+    variable_mc_transformer_flow(pm)
     variable_mc_generation(pm; bounded=false)
+    variable_mc_load(pm)
 
     # Constraints
     constraint_mc_model_current(pm)
 
     for (i,bus) in _PMs.ref(pm, :ref_buses)
-        constraint_mc_theta_ref(pm, i)
+        if typeof(pm)<:LPUBFDiagPowerModel
+            #do nothing, adding theta_ref would lead to redundant constraints
+        else
+            constraint_mc_theta_ref(pm, i)
+        end
```

Comment:
I would recommend condensing this into a single `if` rather than `if`-`else`, e.g.

```julia
if !(typeof(pm)<:LPUBFDiagPowerModel)
    constraint_mc_theta_ref(pm, i)
end
```